### ISSUE TITLE
Quantized _out functions don't follow same conventions as other out functions in the codebase

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#36508]
Currently they're defined like:

```
.op("quantized::add_out(Tensor qa, Tensor qb, Tensor(a!) out)"
     "-> Tensor(a!) out",
    c10::RegisterOperators::options()
      .aliasAnalysis(at::AliasAnalysisKind::FROM_SCHEMA)
      .kernel<QAddOut</*ReLUFused=*/false>>(DispatchKey::QuantizedCPU))
```

However, a standard out function looks like this:

```
- func: angle(Tensor self) -> Tensor
  use_c10_dispatcher: full
  variants: function, method
  supports_named_tensor: True

- func: angle.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
  supports_named_tensor: True
```

cc @kevinbchen (who added the alias analysis annotations to these functions) and @z-a-f (who appears to have originally added these out variants at https://github.com/pytorch/pytorch/pull/23971 )

**Comment by supriyar** [comment]
cc @vkuzo

**Comment by z-a-f** [comment]
The quantized ops live under `torch.quantized.ops.XXX` and are not expected to be used by the end-user in Python.  However the `native_functions` exposes the functions in python (mostly). Do we really need to have identical apis for the two? Also, I am not sure if dispatcher schema supports `*`?

**Comment by dzhulgakov** [comment]
Only kwarg arguments should work for custom op registration api too. Switching to it is probably not BC-compatible in practice but if jit serializes explicit kwargs properly it might be fine

**Comment by ezyang** [comment]
> Do we really need to have identical apis for the two?

That is a great question! This isn't really related to this issue, but I don't know why quantization also registers their operators under the `quantized` namespace. Does anyone else know? (@jerryzh168 ?)

**Comment by andrewor14** [comment]
@jerryzh168 Any context on this? Is it too late to change it now?

**Comment by jerryzh168** [comment]
yeah this is no longer relevant I feel, since these ops are for server/edge CPU use case and used in original eager mode quantization. we have moved on in both backend (more focusing on server GPU + edge) and flow (more focusing on pt2e flow, and the current GPU + triton flow)
